newspaper/parsers.py

```diff
@@ -48,7 +48,17 @@ def get_unicode_html(cls, html):
             return html
         if not html:
             return html
-        converted = UnicodeDammit(html, is_html=True)
+
+        # Guess encoding using "meta" tag.
+        guessed_encodings = None
+
+        document = lxml.html.fromstring(str(html))
+        meta_tag = document.xpath("//meta[@charset] | //meta[@http-equiv='Content-Type']")
+        if meta_tag:
```

Comment:
I think, this part is redundant. UnicodeDammit searches meta charset internally by regular expression, when is_html flag is set. You don't need to parse full document.